File: src/main/Java/boj_step8/boj2292.java
import java.io.*;
import java.util.LinkedList;

public class boj2292 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(reader.readLine());
        int tmp = 6;
        int count = 1;
        LinkedList<Integer> list = new LinkedList<>();
        for (int i = 0; i < n; i++) {
            list.add(count);
            if (list.get(i) >= n) {
                writer.append(String.valueOf(i + 1));
                writer.close();
                return;
            }
            count += tmp;
            tmp += 6;
        }
    }
}

File: src/main/kotlin/others/boj2292.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val n = r.readLine().toInt()
    var tmp = 6
    var temp = 1
    val arr = mutableListOf<Int>()
    for(i in 0 until n) {
        arr.add(temp)
        if(arr[i] >= n) {
            w.write((i + 1).toString())
            w.close()
            return
        }
        temp += tmp
        tmp += 6
    }
}